src/App.tsx

```diff
@@ -1,24 +1,160 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { deleteTodo, getTodos } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorList } from './components/ErrorList/ErrorList';
+import {
+  TodoFilter,
+  FilterTodoStatus,
+} from './components/TodoFilter/TodoFilter';
+import { TodoForm, TodoFormProps } from './components/TodoForm/TodoForm';
 
-const USER_ID = 0;
+const USER_ID = 6848;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [activeTodos, setActiveTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterTodoStatus>(FilterTodoStatus.ALL);
+  const [errors, setErrors] = useState<string>('');
+  const [deletedTodosIds, setDeletedTodosIds] = useState<number[]>([]);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((result) => setTodos(result))
+      .catch(() => setErrors('Unable to get a todo list'));
+  }, []);
+
+  useEffect(() => {
+    const newActiveTodos = todos.filter((todo) => {
+      switch (filter) {
+        case FilterTodoStatus.ACTIVE:
+          return !todo.completed;
+
+        case FilterTodoStatus.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+
+    setActiveTodos(newActiveTodos);
+  }, [todos, filter]);
```

Comment:
This will effect additional rerender. 
Remove additional `activeTodos` state and create const for visible todos. 
NOTE: Filter should call only if `FilterTodoStatus` is not equal `FilterTodoStatus.ALL`.